tortoise/backends/base/schema_generator.py

```diff
@@ -232,34 +208,27 @@ def _get_table_sql(self, model, safe=True) -> dict:
                 fields_with_index.append(db_field)
 
         if model._meta.unique_together:
-            unique_together_sqls = []
-
             for unique_together_list in model._meta.unique_together:
                 unique_together_to_create = []
 
                 for field in unique_together_list:
                     field_object = model._meta.fields_map[field]
 
                     if field_object.source_field:
-                        unique_together_to_create.append(field_object.source_field)
+                        unique_together_to_create.append(self.quote(field_object.source_field))
                     else:
-                        unique_together_to_create.append(field)
-
-                unique_together_sqls.append(
-                    self._get_unique_constraint_sql(unique_together_to_create)
-                )
+                        unique_together_to_create.append(self.quote(field))
 
-            fields_to_create.extend(unique_together_sqls)
+                fields_to_create.append(self._get_unique_constraint_sql(unique_together_to_create))
 
         # Indexes.
-        field_indexes_sqls = [
-            val
-            for val in [
-                self._get_index_sql(model, [field_name], safe=safe)
-                for field_name in fields_with_index
-            ]
-            if val
+        _indexes = [
```

Comment:
May be info about index declared through `index=True` in field kwargs should be returned by field too? 
I'd like to make role of SchemaGenerator as small as possible, so it will be easier to build alternatives for it

Reply:
Yes, I wanted to do that. Because for e.g. case-insensitive or full-text functionality, it can be implemented via collations and indexes on a per-field basis.
The PR was getting colossal enough as is (And I have a lack of time to spend on this), so I wondered if I can defer that to another PR, later?

Or we pull out the non-dependants for 0.14 and release that, and get this perfect first? But considering I spent nearly a month on this, I feel that is a way to delay everything indefinitely.

It is a call we should make, are we going to hold off on features until they are complete, or push partial features in? This is changing as the project matures, and we want to rather have fewer breakages. Are we there yet? 0.14 already introduces some breakages.